pkg/controller/operators/operatorcondition_controller.go

```diff
@@ -119,27 +120,27 @@ func (r *OperatorConditionReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	err = r.syncOperatorConditionStatus(operatorCondition)
+	if err != nil {
+		log.V(1).Error(err, "Error syncing OperatorCondition Status")
+		return ctrl.Result{Requeue: true}, err
```

Comment:
Wouldn't this be requeued because of the error?

Reply:
Correct. I don't see the problem here. If there is an internal error somewhere, we want to requeue to ensure we sync the status properly in the end. Otherwise, the status is not synced and the operator may be stuck in whatever state it may be.

Reply:
Sorry - I might not have been clear. Based on [this documentation](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/reconcile/reconcile.go#L91-L92) I believe that you can just return an error to requeue the request.

Reply:
Agreed, I would just return an error without specifying the `Requeue: true` here.

Reply:
Again, nothing wrong with being explicit either. This is borderline bikeshedding.

Reply:
Fair enough - I don't necessarily disagree, it's just not as intuitive without diving into the vendor code. It looks like the error check is the first conditional check under the hood, and the requeue check is last in the chain, so this is fine with me. It's likely that conditional structure won't change in the future also.